File: Neetcode150/SlidingWindow/LongestSubstringWithoutRepeatingCharacters/LongestSubstringWithoutRepeatingCharacters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        curSet = set()
        longest = 0
        l = 0

        for r in range(len(s)):
            while s[r] in curSet:
                curSet.remove(s[l])
                l += 1
            curSet.add(s[r])
            longest = max(longest, len(curSet))
        
        return longest

                

File: Neetcode150/SlidingWindow/LongestSubstringWithoutRepeatingCharacters/LongestSubstringWithoutRepeatingCharacters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<char> check = new HashSet<char>();
        int longest = 0, l = 0, r = 0;

        while (r < s.Length) {
            if (!check.Contains(s[r])) {
                check.Add(s[r]);
                longest = Math.Max(longest, (r-l+1));
                r++;
            } else {
                check.Remove(s[l]);
                l++;
            }
        }
        return longest;
    }
}
